core/src/main/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentService.java

```diff
@@ -237,8 +237,15 @@ public DetailedEnvironmentV4Response edit(Long workspaceId, String environmentNa
             request.setLocation(locationRequest);
             editRegions(request, environment, cloudRegions);
         }
-        environment = pureSave(environment);
-        return conversionService.convert(environment, DetailedEnvironmentV4Response.class);
+
```

Comment:
I'd prefer the proper solution: env views for stack and cluster and avoiding lazy fetched relations. 
The problem with overusing transactions is that each transaction holds a DB connection during it's lifetime. 
If we keep this attitude to transactions and overuse them, it could cause database access and performance issues. 
I think we should start doing proper design and spend the effort on it, or we might face serious problems later.

Reply:
Follow up issue has been created to cover this bigger effort for the cleaner entity hierarchy.